src/storage-manager/deploy/storage-manager.yaml.template

```diff
@@ -0,0 +1,95 @@
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: storage-manager-ds
+spec:
+  selector:
+    matchLabels:
+      app: storage-manager
+  template:
+    metadata:
+      name: storage-manager
+      labels:
+        app: storage-manager
+      annotations:
+        smbport: "445"
+        nfsport: "2049"
+        localpath: "{{ cluster_cfg["storage-manager"]["localpath"] }}"
+    spec:
+      hostNetwork: false
+      hostname: pai-sm
+      containers:
+      - name: storage-manager
+        image: {{ cluster_cfg["cluster"]["docker-registry"]["prefix"] }}storage-manager:{{ cluster_cfg["cluster"]["docker-registry"]["tag"] }}
+        imagePullPolicy: Always
+        livenessProbe:
+          exec:
+            command:
+            - /usr/bin/check.sh
+          initialDelaySeconds: 60
+          timeoutSeconds: 10
+        securityContext:
+          privileged: True
+        env:
+        - name: PAISMBUSER
+          value: {{ cluster_cfg['storage-manager']['smbuser'] }}
+        - name: PAISMBPWD
+          value: {{ cluster_cfg['storage-manager']['smbpwd'] }}
+        - name: SECURETYPE
+          value: {{ cluster_cfg['storage-manager']['security-type'] }}
+{%- if cluster_cfg["storage-manager"]['security-type'] == "ADS" %}
+        - name: DOMAINUSER
+          value: {{ cluster_cfg['storage-manager']['domainuser'] }}
+        - name: DOMAINPWD
+          value: {{ cluster_cfg['storage-manager']['domainpwd'] }}
+{%- endif %}
+        volumeMounts:
+        - name: pai-storage
+          mountPath: /share/pai
+        - name: pai-config
+          mountPath: /etc/pai-config
+        - name: host-modules
+          mountPath: /lib/modules
+          readOnly: true
+        ports:
+        - containerPort: 445
+          hostPort: 445
+          name: sm-smbport
+        - containerPort: 2049
+          hostPort: 2049
+          name: sm-nfsport
+      volumes:
+        - name: pai-storage
+          hostPath:
```

Comment:
Since it is a hostPath, should we require that machine labelled as "pai-master" should not be labelled as "pai-storage"? Users' data and models are sometimes in a huge size, thus there is some risk if we deploy it to master.

Reply:
Maybe not, if user only have one node to test, he may also use this feature.
Admin should take response for this.